tutorpicasso/commands/run_extra_commands.py

```diff
@@ -0,0 +1,154 @@
+
+    if not picasso_extra_commands:
+        return
+
+    error_message = validate_commands(picasso_extra_commands)
+    if error_message:
+        raise click.ClickException(error_message)
+
+    list(map(run_command, picasso_extra_commands))
+
+
+def validate_commands(commands: Any) -> str:
+    """
+    Takes all the extra commands sent through config.yml and verifies that
+    all the commands are correct before executing them
+
+    Args:
+        commands (list[str] | None): The commands sent through PICASSO_EXTRA_COMMANDS in config.yml
+    """
+    splitted_commands = [
+        split_string(command, COMMAND_CHAINING_OPERATORS) for command in commands
+    ]
+    flat_commands_list: chain[str] = chain.from_iterable(splitted_commands)
+
+    invalid_commands = []
+    misspelled_commands = []
+    for command in flat_commands_list:
+        if "tutor" in command.lower():
+            continue
+
+        if find_tutor_misspelled(command):
+            misspelled_commands.append(command)
+        else:
+            invalid_commands.append(command)
+
+        error_message = ""
+
+        if invalid_commands:
+            error_message += (
+                f"Found some issues with the commands:\n\n"
+                f"=> Invalid commands: {', '.join(invalid_commands)}\n"
+            )
+
+        if misspelled_commands:
+            error_message += (
+                f"=> Misspelled commands: {', '.join(misspelled_commands)}\n"
+            )
+
+        if error_message:
+            error_message += (
+                "Take a look at the official Tutor commands: "
+                "https://docs.tutor.edly.io/reference/cli/index.html"
+            )
+            return error_message
+    return ""
+
+
+def run_command(command: str) -> None:
+    """
+    Run an extra command.
+
+    This method runs the extra command provided.
+
+    Args:
+        command (str): Tutor command.
+    """
+    with subprocess.Popen(
+        command,
+        shell=True,
+        executable="/bin/bash",
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+    ) as process:
+
+        stdout, stderr = process.communicate()
+
+        if process.returncode != 0 or "error" in stderr.lower():
+            raise click.ClickException(
+                f"Command '{command}' failed with return code {process.returncode}. Output: {stdout}. Error: {stderr}"
+            )
+
+        click.echo(stdout)
+
+
+def find_tutor_misspelled(command: str) -> bool:
+    """
+    Look for misspelled occurrences of the word `tutor` in a given string. E.g. ...
+
+    Args:
+        command (str): string to be reviewed.
+
+    Return:
+        True if any misspelled occurrence is found, False otherwise.
+
+    Args:
+        command (str): Command to be reviewed
+
+    Return:
+        If its found the word 'tutor' misspelled is returned True
```

Comment:
Here the return is duplicated. Also, try using the same template (args, return if needed, ... ) for all docstrings.